### PR TITLE
[Inductor] support masked vectorization for the tail_loop for INT8 datatype

### PR DESCRIPTION
**Pull request by jiayisunx**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #128802
* #131745
* __->__ #131155
* #130724

This PR supports masked vectorization for the tail_loop for torch.uint8 and torch.int8 datatype to improve performance.
BTW, I fixed the UT of `byte` by setting the range of the sample inputs  to [0, 255] since the range of `torch.uint8` is [0, 255].

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py
Comment (jgong5) [inline torch/testing/_internal/common_methods_invocations.py:1]: Please explain in the PR description why you need to change it.

**Review by jgong5 — APPROVED** [review]

**Review by leslie-fang-intel — APPROVED** [review]

File: torch/testing/_internal/common_methods_invocations.py
Comment (jiayisunx) [reply torch/testing/_internal/common_methods_invocations.py:1]: Added, thanks!

**Comment by jiayisunx** [comment]
@jansel , could you please review this PR? Thanks!

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**